### ISSUE TITLE
Document NSwag.AssemblyLoader

### ISSUE THREAD

**Issue by barbosatek** [RSuter/NSwag#1058]
I've been digging around the source code and I noticed NSwag.AssemblyLoader does some pretty good level of abstraction from the hosting app domain when loading dlls.

I was wondering if you can document the library and provide usage examples. We would like to use it for loading dlls located on a different folder from its hosting app base directory.

**Comment by RSuter** [comment]
Yes, it was my plan to make this a reusable library eventually (with an own name, independent of nswag). How do you plan to use it?

**Comment by barbosatek** [comment]
Awesome to hear that!

I really like your code because it already supports netcore. I ended up implementing a primitive isolated app runner that only supports dlls (still need support for exe and .config like you package), but this should basically give you an idea on how I plan to use it.

We have a couple tools that load dlls and exes dynamically and execute various tasks as part of our deployment/build pipeline.

Ideally, we want a library that is generic enough so we can reuse it when needed. It should also support netcore and full framework.

These are the signatures and the basic idea I used in my tiny framework.

```
  public class IsolatedAppRunner : IDisposable
  {
    public IsolatedAppRunner(string assemblyPath)
    {
      // Setup new appdomain
      // Setup bin folder
      // Setup AssemblyResolve event for isolate app
    }

    public void Run<T>(params string[] args) where T : IsolatedApp
    {
      // instancite proxy T and call EnvironmentLoaded(params string[] args)
    }

    public void Dispose()
    {
      //Unload appdomain
      //cleanup bin folder
    }
  }
  public abstract class IsolatedApp : MarshalByRefObject
  {
    public Assembly IsolatedAssembly { get; set; }
    public AppDomain IsolatedAppDomain { get; set; }

    public abstract void EnvironmentLoaded(params string[] args);
  }
```
and the usage would be as simple as implementing `IsolatedApp` and executing the following code
```
      using (var isolatedAppRunner = new IsolatedAppRunner(path))
      {
        isolatedAppRunner.Run<MyIsolatedApp>(arg1, "foo", "bar");
      }
```

**Comment by RSuter** [comment]
Currently the core code is not really isolated because it didnt work... i’m not sure if i need the isolation part is really needed in the future.
